app/src/interfaces/input-block-editor/input-block-editor.vue

```diff
@@ -0,0 +1,251 @@
+
+const props = withDefaults(
+	defineProps<{
+		disabled?: boolean;
+		nullable?: boolean;
+		autofocus?: boolean;
+		value?: Record<string, any> | null;
+		bordered?: boolean;
+		placeholder?: string;
+		tools?: string[];
+		folder?: string;
+		font?: 'sans-serif' | 'monospace' | 'serif';
+	}>(),
+	{
+		disabled: false,
+		nullable: false,
+		autofocus: false,
+		value: () => null,
+		bordered: true,
+		tools: () => ['header', 'nestedlist', 'code', 'image', 'paragraph', 'checklist', 'quote', 'underline'],
+		font: 'sans-serif',
+	}
+);
+
+const emit = defineEmits(['input']);
+
+const { t } = useI18n();
+
+const collectionStore = useCollectionsStore();
+
+const { currentPreview, setCurrentPreview, fileHandler, setFileHandler, unsetFileHandler, handleFile } =
+	useFileHandler();
+
+const editorjsRef = ref<EditorJS>();
+const uploaderComponentElement = ref<HTMLElement>();
+const editorElement = ref<HTMLElement>();
+const haveFilesAccess = Boolean(collectionStore.getCollection('directus_files'));
+const haveValuesChanged = ref<boolean>(false);
+
+const tools = getTools(
+	{
+		addTokenToURL,
+		baseURL: api.defaults.baseURL,
+		setFileHandler,
+		setCurrentPreview,
+		getUploadFieldElement: () => uploaderComponentElement,
+	},
+	props.tools,
+	haveFilesAccess
+);
+
+onMounted(async () => {
+	const sanitizedValue = sanitizeValue(props.value);
+
+	editorjsRef.value = new EditorJS({
+		logLevel: 'ERROR' as EditorJS.LogLevels,
+		holder: editorElement.value,
+		readOnly: false,
+		placeholder: props.placeholder,
+		minHeight: 72,
+		onChange: (api: any, event: any) => emitValue(api, event),
+		tools: tools,
+	});
+
+	// we have initial data, so we render it once the editor is ready...
+	await editorjsRef.value.isReady;
+
+	if (sanitizedValue) {
+		await editorjsRef.value.render(sanitizedValue);
+	}
+
+	if (props.autofocus) {
+		editorjsRef.value.focus();
+	}
+});
+
+onUnmounted(() => {
+	if (!editorjsRef.value) return;
+
+	editorjsRef.value.destroy();
+});
+
+watch(
+	() => props.value,
+	async (newVal: any, oldVal: any) => {
+		if (!editorjsRef.value || !editorjsRef.value.isReady || haveValuesChanged.value) return;
+
+		if (fileHandler.value !== null) return;
+
+		if (isEqual(newVal?.blocks, oldVal?.blocks)) return;
+
+		try {
+			await editorjsRef.value.isReady;
+			const sanitizedValue = sanitizeValue(newVal);
+
+			if (sanitizedValue) {
+				await editorjsRef.value.render(sanitizedValue);
+			} else {
+				editorjsRef.value.clear();
+			}
+		} catch (err: any) {
+			unexpectedError(err);
+		}
+
+		haveValuesChanged.value = false;
+	}
+);
+
+async function emitValue(context: EditorJS.API, _event: CustomEvent) {
+	if (props.disabled || !context || !context.saver) return;
+	haveValuesChanged.value = true;
+
+	try {
+		const result: EditorJS.OutputData = await context.saver.save();
+
+		if (!result || result.blocks.length < 1) {
+			emit('input', props.nullable ? null : '{}');
+			return;
+		}
```

Comment:
```suggestion
		if (!result || result.blocks.length < 1) {
			emit('input', null);
			return;
		}
```

Can we remove the `nullable` prop? 

The returning of `"{}"` results in type warnings as the type for `value` is `Record<string, any> | null`.

Reply:
I think this makes sense 👍 Will proceed to implement this.